lib/route.js

```diff
@@ -294,7 +294,7 @@ function buildRouting (options) {
     if (trustProxy) {
       ip = proxyAddr(req, proxyFn)
       ips = proxyAddr.all(req, proxyFn)
-      if (ip !== undefined && req.headers['x-forwarded-host']) {
+      if (proxyFn(req.connection.remoteAddress, 0) && req.headers['x-forwarded-host']) {
```

Comment:
We are already calling proxyFn in the two lines above. Maybe we should check ips instead?

Reply:
Correct me if I'm wrong, 
IPs returning Array of the request socket connection IP
And if that a trusted IP he will contact the X-Forwarded-For IPs 

So I don't know if there's a way to use that value 
proxyFn returning just a boolean

Reply:
From the docs, having ips.length === 0 (or maybe 1) will achieve the same result. It should stop including ip addresses at the first untrusted

Reply:
The ips.length will be minimum 1  (socket connection IP)
If the IP is trusted and the proxy sends X-Forwarded-For headers the IPs length will be greater.

**but in a situation** where the trusted proxy for some reason does not supply 
X-Forwarded-For Header The IPs length will still be 1.

Reply:
I don’t see that tested in this PR, i.e. not what you are trying to fix here. Can you add a test?

I don’t understand because we are already calling that function 2+ times for every request, and I would really would like to avoid calling it more if possible.